firebase-authexchange/src/main/kotlin/com/google/firebase/authexchange/network/Routes.kt

```diff
@@ -0,0 +1,103 @@
+package com.google.firebase.authexchange.network
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Defines what an HTTP request may look like when interacting with the backend API for
```

Comment:
> You can be more direct in the doc:
> 
> `A request to the backend`
> 
> and then, below
> 
> `A response from the backend`

Honestly, I would pitch otherwise. While I think the goal (especially on the first line) should be to be as direct as possible- being too brief and assuming a common understanding often-times leads to a misunderstanding. I've personally ran into it before, where there's been some documentation that assumed a common understanding that I lacked- because my process of learning more was starting at the code and working up :')
Curious to hear your thoughts though

> I'm curious why are you defining them as sealed interfaces.

Because they can all be declared at compile time (people won't be extending them outside this file). Additionally, it comes with the benefit of additional compiler optimizations, and exhausted casting. So we can do stuff like this:

```kotlin
when(request) {
 ExchangeInstallationAuthTokenRequestP -> {}
 ExchangeCustomTokenRequestP -> {}
 ExchangeOidcTokenRequestP -> {}
}
```

And the compiler recognizes that this is all the possible combinations, so there's not need for an `else` statement that just throws an error. There's a couple other benefits that've gotta do with type casting- but those are what come to mind initially.

> Also, why internal.

Because you can't hide packages like in java with `package-info`. You would have to manually exclude each file. But more importantly, it lets folks know if they should ever try to accidentally use one of these classes- that it's intended for internal use only. That is, it's much easier for developers to see that they should use `com.google.firebase.authexchange.AuthExchangeToken` instead of `com.google.firebase.authexchange.network.routes.AuthExchangeToken`, because one is explicitly marked as `internal` wheres the other one is just public.